projects/groprotocol/helpers.js

```diff
@@ -0,0 +1,236 @@
+  stateMutability: "view",
+  type: "function",
+};
+
+/* lpPositions:{
+    balance,
+    token
+}[]
+*/
+async function unwrapUniswapLPs(
+  balances,
+  lpPositions,
+  block,
+  chain = "ethereum",
+  transformAddress = (addr) => addr,
+  excludeTokensRaw = [],
+  retry = false
+) {
+  const excludeTokens = excludeTokensRaw.map((addr) => addr.toLowerCase());
+  const lpTokenCalls = lpPositions.map((lpPosition) => ({
+    target: lpPosition.token,
+  }));
+  const lpReserves = sdk.api.abi.multiCall({
+    block,
+    abi: lpReservesAbi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  const lpSupplies = sdk.api.abi.multiCall({
+    block,
+    abi: lpSuppliesAbi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  const tokens0 = sdk.api.abi.multiCall({
+    block,
+    abi: token0Abi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  const tokens1 = sdk.api.abi.multiCall({
+    block,
+    abi: token1Abi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  if (retry) {
+    await Promise.all(
+      [
+        [lpReserves, lpReservesAbi],
+        [lpSupplies, lpSuppliesAbi],
+        [tokens0, token0Abi],
+        [tokens1, token1Abi],
+      ].map(async (call) => {
+        await requery(await call[0], chain, block, call[1]);
+      })
+    );
+  }
+  await Promise.all(
+    lpPositions.map(async (lpPosition) => {
+      try {
+        const lpToken = lpPosition.token;
+        const token0 = (await tokens0).output
+          .find((call) => call.input.target === lpToken)
+          .output.toLowerCase();
+        const token1 = (await tokens1).output
+          .find((call) => call.input.target === lpToken)
+          .output.toLowerCase();
+        const supply = (await lpSupplies).output.find(
+          (call) => call.input.target === lpToken
+        ).output;
+        const { _reserve0, _reserve1 } = (await lpReserves).output.find(
+          (call) => call.input.target === lpToken
+        ).output;
+        if (!excludeTokens.includes(token0)) {
+          const token0Balance = BigNumber(lpPosition.balance)
+            .times(BigNumber(_reserve0))
+            .div(BigNumber(supply));
+          sdk.util.sumSingleBalance(
+            balances,
+            await transformAddress(token0),
+            token0Balance.toFixed(0)
+          );
+        }
+        if (!excludeTokens.includes(token1)) {
+          const token1Balance = BigNumber(lpPosition.balance)
+            .times(BigNumber(_reserve1))
+            .div(BigNumber(supply));
+          sdk.util.sumSingleBalance(
+            balances,
+            await transformAddress(token1),
+            token1Balance.toFixed(0)
+          );
+        }
+      } catch (e) {
+        console.log(
+          `Failed to get data for LP token at ${lpPosition.token} on chain ${chain}`
+        );
+        throw e;
+      }
+    })
+  );
+}
+
+const crvPools = {
+  // PWRD-3CRV
+  "0xbcb91e689114b9cc865ad7871845c95241df4105": {
+    swapContract: "0x001C249c09090D79Dc350A286247479F08c7aaD7",
```

Comment:
maybe we can resuse core `unwrapCrv` helper if we set `swapContract: '0xbcb91e689114b9cc865ad7871845c95241df4105'` and add exclusion logic in the core function. wdyt @0xngmi ?

Reply:
could do it but if you prefer to keep it here i'm cool with that too